datafusion/common/src/scalar.rs

```diff
@@ -618,6 +619,28 @@ impl ScalarValue {
             precision, scale
         )))
     }
+
+    /// Returns a [`ScalarValue::IntervalYearMonth`] representing
+    /// `years` years and `months` months
```

Comment:
These seem very nice to have. I think more convenience functions like this are the key to terser and more readable code.